jane/doc/extensions/modes/reference.md

```diff
@@ -0,0 +1,42 @@
+The goal of this document is to be a reasonably complete reference to the mode system in
+OCaml.
+
+<!-- CR zqian: For a gentler introduction, see [the introduction](intro.md). -->
+
+The mode system in the compiler tracks various properties of values, so that certain
+performance-enhancing operations can be performed safely. For example:
+- Locality tracks escaping. See [the local allocations reference](../local/reference.md)
+- Uniqueness and linearity tracks aliasing. See [the uniqueness reference](../uniqueness/reference.md)
+- Portability and contention tracks inter-thread sharing.
+    <!-- CR zqian: reference for portability and contention -->
+
+# Lazy
+`lazy e` contains a thunk that evaluates `e`, as well as a mutable cell to store the
+result of `e`. Upon construction, the mode of `lazy e` is weaker than `e`. For example, if
+`e` is `nonportable`, then `lazy e` cannot be `portable`. Upon destruction (forcing a lazy
+value), the result is weaker than the mode of lazy value. For example, forcing a
+`nonportable` lazy value cannot give a `portable` result. Additionally, forcing a lazy
+value involves accessing the mutable cell and thus requires the lazy value to be
+`uncontended`.
+
+Currently, the above rules don't apply to the locality axis, because both the result and
+the lazy value are heap-allocated, so they are always `global`.
+
+Additionally, upon construction, the comonadic fragment of `lazy e` is weaker than the
+thunk. The thunk is checked as `fun () -> e`, potentially closing over variables, which
+weakens its comonadic fragment. This rule doesn't apply to several axes:
```

Comment:
```suggestion
may affect its comonadic modes. This rule doesn't apply to several axes:
```

Reply:
Hmm, "weaken" is more precise than "affect", right?

Reply:
Sure. But it looks like you've made this change.